tests/test_mirrors.py

```diff
@@ -58,21 +58,29 @@ def setUp(self):
                  'confined_target_dirs' : ['targets/release/',
                                             'targets/release/']},
      'mirror3': {'url_prefix' : 'http://mirror3.com',
-                 'metadata_path' : 'metadata',
                  'targets_path' : 'targets',
-                 'confined_target_dirs' : ['targets/release/',
-                                            'targets/release/']}}
+                 'confined_target_dirs' : ['targets/release/v2/']},
```

Comment:
Why did you have to change that?

Reply:
ah, that could maybe have been a separate change: there was nothing testing if confinement matches subdirectories (it does not). 
The test I added (one with 2 matches) tests that while it also makes sure things work without a metadata_path.

Reply:
> The test I added (one with 2 matches) tests that while it also makes sure things work without a metadata_path.

It tests that you have added makes sure things works without a metadata_path, yes, but you haven't tested if confinement matches subdirectories.
I don't think we need such a test, because to check if a directory is confinement in a list is a function of securesyslib and it's tested there:
https://github.com/secure-systems-lab/securesystemslib/blob/94d83b943f2bcc020619547d83cc037bd78dfb61/tests/test_util.py#L165

Reply:
> It tests that you have added makes sure things work without a metadata_path, yes, but you haven't tested if confinement matches subdirectories.

If subdirectories did match, mirror2 would be in the results -> test would fail.

> I don't think we need such a test, because to check if a directory is confinement in a list is a function of securesystemslib

That's an implementation detail of `get_list_of_mirrors()`... if we change the implementation (or if securesystemslib changes their implementation) we'd want to know if results change.

In any case: I'm not adding a confinement test here: I'm adding a test for missing metadata_path that also happens to check that confinement works they way it was designed to.

Reply:
> If subdirectories did match, mirror2 would be in the results -> test would fail.

Yes, you are right.
Maybe it will be even better if you add a comment that subdirectories are not expected to match?
It's possible that this is well documented on multiple places.
That's a NIT and it's your decision. 

> That's an implementation detail of get_list_of_mirrors()... if we change the implementation (or if securesystemslib changes their implementation) we'd want to know if results change.
> In any case: I'm not adding a confinement test here: I'm adding a test for missing metadata_path that also happens to check that confinement works they way it was designed to.

Yes, you are right.